java/client/src/test/java/glide/api/RedisClientTest.java

```diff
@@ -2720,6 +2721,29 @@ public void time_returns_success() {
         assertEquals(payload, response.get());
     }
 
+    @SneakyThrows
+    @Test
+    public void sinter_returns_success() {
+        // setup
+        String[] keys = new String[] {"key1", "key2"};
+        Set<String> value = Set.of("1", "2");
+
+        CompletableFuture<Set<String>> testResponse = new CompletableFuture<>();
+        testResponse.complete(value);
+
+        // match on protobuf request
+        when(commandManager.<Set<String>>submitNewCommand(eq(SInter), eq(keys), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<Set<String>> response = service.sinter(keys);
+        Set<String> payload = response.get();
+
+        // verify
+        assertEquals(testResponse, response);
+        assertEquals(payload, response.get());
```

Comment:
```suggestion
        assertEquals(payload, value);
```

shouldn't it be like this?

Reply:
Fixed, thanks